workspace/__lib__/xod-dev/text-lcd/print-at(text-lcd-i2c-device)/patch.cpp

```diff
@@ -0,0 +1,37 @@
+
+struct State {
+};
+
+{{ GENERATED_CODE }}
+
+void printAt(LiquidCrystal_I2C* lcd, uint8_t rowIndex, uint8_t posIndex, uint8_t len, XString str) {
+    lcd->setCursor(posIndex, rowIndex);
+    uint8_t whitespace = len;
+    for (auto it = str.iterate(); it && whitespace > 0; ++it, --whitespace)
+        lcd->write(*it);
+
+    // Clear the rest of the area
+    while (whitespace--)
+        lcd->write(' ');
+}
+
+void evaluate(Context ctx) {
+    auto t = getValue<input_DEV>(ctx);
+
+    if (isInputDirty<input_DO>(ctx)) {
+        XString str = getValue<input_VAL>(ctx);
+        uint8_t row = (uint8_t) getValue<input_ROW>(ctx);
+        uint8_t pos = (uint8_t) getValue<input_POS>(ctx);
+        uint8_t len = (uint8_t) getValue<input_LEN>(ctx);
+
+        if (row < 0 || row >= t.rows || pos < 0 || pos >= t.cols || pos + len > t.cols) {
+            raiseError<output_OK>(ctx);
```

Comment:
I don’t like the idea of erroring if `pos + len > t.cols`. A more humane behavior would be trimming the length to fit the right boundary of the display.